packages/stats/gbstats/bayesian/tests.py

```diff
@@ -223,3 +227,114 @@ def get_risk(mu, sigma) -> List[float]:
         risk_ctrl = float((1.0 - prob_ctrl_is_better) * mn_pos)
         risk_trt = -float(prob_ctrl_is_better * mn_neg)
         return [risk_ctrl, risk_trt]
+
+
+@dataclass
+class BanditConfig(BayesianConfig):
+    burn: int = 0
+    iteration: int = 0
+    top_two: bool = True
+    prior_distribution: GaussianPrior = field(default_factory=GaussianPrior)
+
+
+class Bandits(object):
+    def __init__(
+        self,
+        stat: Sequence[Union[ProportionStatistic, SampleMeanStatistic]],
+        config: BanditConfig,
+    ):
+        self.stat = stat
+        self.config = config
+        self.n_variations = len(stat)
+
+    @property
+    def variation_means(self):
+        return np.array([arm.mean for arm in self.stat])
+
+    @property
+    def variation_variances(self):
+        return np.array([arm.variance for arm in self.stat])
+
+    @property
+    def variation_counts(self):
+        return np.array([arm.n for arm in self.stat])
+
+    @property
+    def prior_precision(self):
+        return np.full(
+            (self.n_variations,),
+            int(self.config.prior_distribution.proper)
+            / self.config.prior_distribution.variance,
+        )
+
+    @property
+    def data_precision(self):
+        return np.array(
+            [
+                float(n) / v if v > 0 else 0
+                for n, v in zip(self.variation_counts, self.variation_variances)
+            ]
+        )
+
+    @property
+    def posterior_precision(self):
+        return self.prior_precision + self.data_precision
+
+    @property
+    def posterior_variance(self):
+        return 1 / self.posterior_precision
+
+    @property
+    def prior_mean(self):
+        return np.full((self.n_variations,), self.config.prior_distribution.mean)
+
+    @property
+    def posterior_mean(self):
+        return self.posterior_variance * (
+            self.prior_precision * self.prior_mean
+            + self.data_precision * self.variation_means
+        )
+
+    # number of Monte Carlo samples to perform when sampling to estimate weights for the SDK
+    @property
+    def n_samples(self):
+        return int(1e4)
+
+    # function that computes thompson sampling variation weights
+    def compute_variation_weights(self):
```

Comment:
Eventually we'll need to do something like include `p` in some kind of `BanditResult` object that will be consumed by the back-end.

Reply:
I have created class `BanditWeights` that has the weights and a string indicating whether or not the weights are updated.  I incorporated a `bandit_weights` attribute of type BanditWeights into DimensionResponse.  I created a unit test that indicates how this can flow through `format_results`.